spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/DefaultChannelFactory.java

```diff
@@ -34,17 +34,23 @@
 	MessageConverterConfigurer messageConverterConfigurer;
 
 	@Override
-	public MessageChannel createChannel(String name, Class<?> inputChannelType) throws Exception {
-		MessageChannel messageChannel = createMessageChannel(inputChannelType);
+	public <T extends MessageChannel> T createBindableChannel(String name, Class<T> channelType)
+			throws Exception {
+		T messageChannel = createMessageChannel(channelType);
 		messageConverterConfigurer.configureMessageConverters(messageChannel, name);
 		return messageChannel;
 	}
 
-	private MessageChannel createMessageChannel(Class<?> messageChannelType) {
-		return isPollable(messageChannelType) ? new QueueChannel() : new DirectChannel();
+	@Override
+	public <T extends MessageChannel> T createSharedChannel(Class<T> channelType) throws Exception {
+		return createMessageChannel(channelType);
+	}
+
+	private <T extends MessageChannel> T createMessageChannel(Class<T> messageChannelType) {
+		return isPollable(messageChannelType) ? (T) new QueueChannel() : (T) new DirectChannel();
```

Comment:
this makes me think we should only expect `SubscribableChannel` or `PollableChannel` after all... or do we actually need separate methods for pollable and subscribable? cc @mbogoevici